src/wallet/api/wallet.cpp

```diff
@@ -923,9 +944,12 @@ bool WalletImpl::connectToDaemon()
 
 Wallet::ConnectionStatus WalletImpl::connected() const
 {
+    // check if asynchronous wallet close started
+    if(m_is_shutting_down)
+        return Wallet::ConnectionStatus_Disconnected;
     uint32_t version = 0;
-    bool is_connected = m_wallet->check_connection(&version);
-    if (!is_connected)
+    m_is_connected = m_wallet->check_connection(&version);
```

Comment:
Is this an expensive function call? Is it necessary aliasing the value desireable?